velox/common/base/Fs.h

```diff
@@ -15,10 +15,32 @@
  */
 #pragma once
 
+#include <optional>
+
 #if __has_include("filesystem")
 #include <filesystem>
 namespace fs = std::filesystem;
 #else
 #include <experimental/filesystem>
 namespace fs = std::experimental::filesystem;
 #endif
+
+namespace facebook::velox::common {
+
+/// Generates a file directory specified by 'dirPath'. The generation will be
+/// recursive. Non-exist parent directories will also be created. Returns true
+/// if creation is successful, false otherwise. Error message will be printed if
+/// creation is unsuccessful, but already created directories will not be
+/// removed.
+bool generateFileDirectory(const char* dirPath);
```

Comment:
how different is this function from

> https://en.cppreference.com/w/cpp/filesystem/create_directory

?

Reply:
This function uses fs::create_directories() to achieve its purpose. The additional logic this function provides is extra error handling, so that caller does not need to handle detailed file system errors. 


For example here when error code is 0 the fs::create_directories() will result unsuccessful but what it actually means is the directory is already there, just that the action of creating the directory was not performed. From the caller's perspective it should be successful because the directory is already there ready for use.